Gordon360/Services/HousingService.cs

```diff
@@ -0,0 +1,774 @@
+
+            IEnumerable<GET_AA_APARTMENT_CHOICES_BY_APP_ID_Result> existingApartmentChoiceResult = null;
+
+            appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
+
+            // Get the apartment preferences that are already stored in the database for this application
+            existingApartmentChoiceResult = RawSqlQuery<GET_AA_APARTMENT_CHOICES_BY_APP_ID_Result>.query("GET_AA_APARTMENT_CHOICES_BY_APP_ID @APPLICATION_ID", appIDParam);
+            if (existingApartmentChoiceResult == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The hall information could not be found." };
+            }
+
+            // List of apartment choices that are in the array recieved from the frontend but not yet in the database
+            List<ApartmentChoiceViewModel> apartmentChoicesToAdd = new List<ApartmentChoiceViewModel>(newApartmentChoices);
+
+            // List of apartment choices that are in both the array recieved from the frontend and the database
+            List<ApartmentChoiceViewModel> apartmentChoicesToUpdate = new List<ApartmentChoiceViewModel>();
+
+            // List of apartment choices that are in the database but not in the array recieved from the frontend
+            List<ApartmentChoiceViewModel> apartmentChoicesToRemove = new List<ApartmentChoiceViewModel>();
+
+            // Check whether any apartment choices were found matching the given application ID number
+            if (existingApartmentChoiceResult.Any())
+            {
+                foreach (GET_AA_APARTMENT_CHOICES_BY_APP_ID_Result existingApartmentChoice in existingApartmentChoiceResult)
+                {
+                    ApartmentChoiceViewModel newMatchingApartmentChoice = null;
+                    newMatchingApartmentChoice = newApartmentChoices.FirstOrDefault(x => x.HallName == existingApartmentChoice.BLDG_CDE);
+                    if (newMatchingApartmentChoice != null)
+                    {
+                        // If the apartment is in both the new apartment list and the existing apartment list, then we do NOT need to add it to the database
+                        apartmentChoicesToAdd.Remove(newMatchingApartmentChoice);
+                        if (newMatchingApartmentChoice.HallRank != existingApartmentChoice.Ranking)
+                        {
+                            // If the apartment is in both the new and existing apartment lists but has different ranking values, then we need to update that in the database
+                            apartmentChoicesToUpdate.Add(newMatchingApartmentChoice);
+                        }
+                    }
+                    else
+                    {
+                        // If the apartment is in the existing list but not in the new list of apartments, then we need to remove it from the database
+                        apartmentChoicesToRemove.Add(newMatchingApartmentChoice);
```

Comment:
This is broken in the same way as the above issue with applicantsToRemove. 

It makes it impossible to remove an apartment choice from an existing application currently.